lib/opentok/archives.rb

```diff
@@ -96,11 +96,13 @@ def find(archive_id)
     #   the most recent archive. If you do not specify an offset, 0 is used.
     # @option options [integer] :count Optional. The number of archives to be returned. The maximum
     #   number of archives returned is 1000.
+    # @option options [String] :session_id Optional. The session ID that archives belong to. This is
+    #   useful when listing multiple archives for a {https://tokbox.com/developer/guides/archiving/#automatic-archives automatically archived session}
     #
     # @return [ArchiveList] An ArchiveList object, which is an array of Archive objects.
     def all(options = {})
-      raise ArgumentError, "Limit is invalid" unless options[:count].nil? or (0..100).include? options[:count]
-      archive_list_json = @client.list_archives(options[:offset], options[:count])
+      raise ArgumentError, "Limit is invalid" unless options[:count].nil? or (0..1000).include? options[:count]
```

Comment:
If I understand correctly, the total number of archives returned could be up to 1000.